lua/harpoon/tabline.lua

```diff
@@ -0,0 +1,80 @@
+local Dev = require("harpoon.dev")
+local log = Dev.log
+
+local M = {}
+
+function get_color(group, attr)
+  return vim.fn.synIDattr(vim.fn.synIDtrans(vim.fn.hlID(group)), attr)
+end
+
+
+local function shorten_filenames(filenames)
+  local shortened = {}
+
+  local counts = {}
+  for _, file in ipairs(filenames) do
+    local name = vim.fn.fnamemodify(file.filename, ":t")
+    counts[name] = (counts[name] or 0) + 1
+  end
+
+  for _, file in ipairs(filenames) do
+    local name = vim.fn.fnamemodify(file.filename, ":t")
+
+    if counts[name] == 1 then
+      table.insert(shortened, { filename = vim.fn.fnamemodify(name, ":t") })
+    else
+      table.insert(shortened, { filename = file.filename })
+    end
+  end
+
+  return shortened
+end
+
+function M.setup(opts)
+    function _G.tabline()
```

Comment:
what is this?  i don't understand _G

Reply:
we need this to call inside tabline, like: 

```
vim.o.tabline = '%!v:lua.tabline()'
```